### ISSUE TITLE
`Image` type is not an instance of `Writeable`

### ISSUE THREAD

**Issue by mherzl** [LaurentRDC/hakyll-images#1]
As described in [this](https://stackoverflow.com/questions/54274208/how-to-reconcile-types-between-hakyll-and-hakyll-images) stackoverflow question, the type `Image` (defined in `hakyll-images`'s [`Hakyll.Images` module](https://hackage.haskell.org/package/hakyll-images-0.3.1/docs/Hakyll-Images.html#t:Image)), is not an instance of `Writeable` (defined in `hakyll`'s [`Hakyll.Core.Writable` module](http://hackage.haskell.org/package/hakyll-4.12.5.0/docs/Hakyll-Core-Writable.html) as required by `hakyll`'s `compile` (defined in `hakyll`'s [`Hakyll.Core.Rules` module](http://hackage.haskell.org/package/hakyll-4.12.5.0/docs/Hakyll-Core-Rules.html#v:compile)).

Due to the missing `Writeable` class for `Image`, the following example from `hakyll-images`'s [`README.md`](https://github.com/LaurentRDC/hakyll-images#readme) does not compile.

```
import Hakyll
import Hakyll.Images        ( loadImage
                            , scaleImageCompiler
                            )
main = hakyll $ do
    -- Scale images to fit within a 600x400 box
    -- Aspect ratio will be preserved
    match "images/*" $ do
        route idRoute
        compile $ loadImage
            >>= scaleImageCompiler 600 400
```
Instead, it throws a 'no instance' error:

> No instance for (Writable
>                          hakyll-images-0.3.1:Hakyll.Images.Common.Image)
>         arising from a use of ‘compile’

What is the solution to this? Should a `Writeable` instance for `Image` be added?

**Comment by LaurentRDC** [comment]
@mherzl Unfortunately, this bug made its way into the hakyll-images 0.3.1 release. It was fixed subsequently in hakyll-images 0.4 and above. Simply update to the latest version to get rid of this problem.

I forgot to add a test that checked the README code. It has been added as of version 0.4. Please forgive me for the large oversight.

**Comment by mherzl** [comment]
Thank you for the reply and explanation! Yep somehow I was using an older version, that example is compiling now. :)
